controller/utils.go

```diff
@@ -13,6 +15,73 @@ import (
 	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
 )
 
+type ExponentialBackoff struct {
+	interval           map[string]time.Duration
+	lastAttempt        map[string]time.Time
+	maxBackoffInterval time.Duration
+	mu                 sync.Mutex
+}
```

Comment:
@bachmanity1 To simplify the implemation and improve the maintainability, is it possible to replace the `ExponentialBackoff` implementation with client-go's `flowcontrol.Backoff`?

Reply:
As I mentioned earlier, it’s hard to decide when to reset the backoff because even when the pod creation request succeeds, the pod may still fail to start. We need to monitor the pod for some time to determine if it’s running, so that we can reset the backoff. The question is, how long should we monitor the pod?

Reply:
@derekbit  Reading the `flowcontrol.Backoff` docs again, I realized it actually does have the functionality I was talking about. We don't need to explicitly reset the backoff, we can just call [GC](https://github.com/kubernetes/client-go/blob/v0.33.2/util/flowcontrol/backoff.go#L149) periodically to clean up expired records. So with a small util function, we can get `flowcontrol.Backoff` to behave the way we want.

The only issue is that `flowcontrol.Backoff` doesn't support dynamically updating the max backoff. But I don't think many people would need to change the default anyway, so how about just making the max backoff a constant (like 2 minutes?) and just using `flowcontrol.Backoff` like you originally suggested?